Tests/SentryTests/Networking/SentryHTTPInterceptorTests.swift

```diff
@@ -0,0 +1,256 @@
+        let request = URLRequest(url: URL(string: SentryHTTPInterceptorTests.httpUrl)!)
+        XCTAssertTrue(SentryHttpInterceptor.canInit(with: request))
+    }
+    
+    func testCanInitWithRequestHTTPS() {
+        let request = URLRequest(url: URL(string: SentryHTTPInterceptorTests.httpsUrl)!)
+        XCTAssertTrue(SentryHttpInterceptor.canInit(with: request))
+    }
+    
+    func testCannotInitWithRequestWS() {
+        let request = URLRequest(url: URL(string: SentryHTTPInterceptorTests.wsUrl)!)
+        XCTAssertFalse(SentryHttpInterceptor.canInit(with: request))
+    }
+    
+    func testCannotInitWithSentryRequest() {
+        let request = URLRequest(url: URL(string: fixture.options.dsn!)!)
+        XCTAssertFalse(SentryHttpInterceptor.canInit(with: request))
+    }
+    
+    func testCannotInitWithoutScopeTransaction() {
+        fixture.scope.span = nil
+        let request = URLRequest(url: URL(string: SentryHTTPInterceptorTests.httpsUrl)!)
+        XCTAssertFalse(SentryHttpInterceptor.canInit(with: request))
+    }
+    
+    func testCanonicalRequestForRequest() {
+        let request = URLRequest(url: URL(string: SentryHTTPInterceptorTests.httpUrl)!)
+        let newRequest = SentryHttpInterceptor.canonicalRequest(for: request)
+        
+        let spanHeader = fixture.span.toTraceHeader()
+        XCTAssertNotEqual(request, newRequest)
+        XCTAssertEqual(spanHeader.value(), newRequest.value(forHTTPHeaderField: SENTRY_TRACE_HEADER))
+        
+        let interceptFlag = URLProtocol.property(forKey: SENTRY_INTERCEPTED_REQUEST, in: newRequest) as? NSNumber
+        
+        XCTAssertTrue(interceptFlag!.boolValue)
+    }
+    
+    func testCanonicalRequestForRequestWithoutTransaction() {
+        fixture.scope.span = nil
+        let request = URLRequest(url: URL(string: SentryHTTPInterceptorTests.httpUrl)!)
+        let newRequest = SentryHttpInterceptor.canonicalRequest(for: request)
+        
+        XCTAssertEqual(request, newRequest)
+        XCTAssertNil(newRequest.value(forHTTPHeaderField: SENTRY_TRACE_HEADER))
+    }
+    
+    func testIgnoreAlreadyInterceptedRequest() {
+        let request = URLRequest(url: URL(string: SentryHTTPInterceptorTests.httpUrl)!)
+        let newRequest = SentryHttpInterceptor.canonicalRequest(for: request)
+        XCTAssertFalse(SentryHttpInterceptor.canInit(with: newRequest))
+    }
+    
+    func testProtocolGetRegisteredAndUnregistered() {
+        NSURLProtocolSwizzle.swizzleURLProtocol()
+        NSURLProtocolSwizzle.shared.registerCallback = { protocolClass in
+            XCTAssertTrue(protocolClass === SentryHttpInterceptor.self)
+        }
+        NSURLProtocolSwizzle.shared.unregisterCallback = { protocolClass in
+            XCTAssertTrue(protocolClass === SentryHttpInterceptor.self)
+        }
+        let integration = SentryNetworkTrackingIntegration()
+        integration.install(with: fixture.options)
+        integration.uninstall()
```

Comment:
`m`: Removing `NSURLProtocol registerClass` and `NSURLProtocol unregisterClass` from `SentryNetworkTrackingIntegration` doesn't make this test fail. We need to make sure that the callbacks are also called.

```suggestion
        
        var registerCallbackCalled = false
        NSURLProtocolSwizzle.shared.registerCallback = { protocolClass in
            registerCallbackCalled = true
            XCTAssertTrue(protocolClass === SentryHttpInterceptor.self)
        }
        
        var unregisterCallbackCalled = false
        NSURLProtocolSwizzle.shared.unregisterCallback = { protocolClass in
            unregisterCallbackCalled = true
            XCTAssertTrue(protocolClass === SentryHttpInterceptor.self)
        }
        let integration = SentryNetworkTrackingIntegration()
        integration.install(with: fixture.options)
        XCTAssertTrue(registerCallbackCalled)
        
        integration.uninstall()
        XCTAssertTrue(unregisterCallbackCalled)
```